File: InJava/app/src/main/java/com/barryzeha/pomodoroapp/model/adapters/ViewPagerAdapter.java
package com.barryzeha.pomodoroapp.model.adapters;

import androidx.annotation.NonNull;
import androidx.fragment.app.Fragment;
import androidx.fragment.app.FragmentManager;

import androidx.lifecycle.Lifecycle;
import androidx.viewpager2.adapter.FragmentStateAdapter;

import com.barryzeha.pomodoroapp.view.HistoryFragment;
import com.barryzeha.pomodoroapp.view.SettingsFragment;
import com.barryzeha.pomodoroapp.view.TimerFragment;

/****
 * Project PomodoroApp
 * Created by Barry Zea H. on 18/10/2022
 * Copyright (c)  All rights reserved.
 ***/
public class ViewPagerAdapter extends FragmentStateAdapter {

    public ViewPagerAdapter(FragmentManager fragmentManger, Lifecycle lifecycle){
        super(fragmentManger,lifecycle);
    }

    @NonNull
    @Override
    public Fragment createFragment(int position) {
        Fragment fragment;
        switch(position){
            case 1:
                fragment=new HistoryFragment();
                break;
            case 2:
                fragment= new SettingsFragment();
                break;
            default :
                fragment = new TimerFragment();
                break;
        }
        return fragment;
    }

    @Override
    public int getItemCount() {
        return 3;
    }
}

File: InKotlin/app/src/main/java/com/barryzeha/pomodoroapp/model/adapters/ViewPagerAdapter.kt
package com.barryzeha.pomodoroapp.model.adapters

import androidx.fragment.app.Fragment
import androidx.fragment.app.FragmentManager

import androidx.lifecycle.Lifecycle
import androidx.viewpager2.adapter.FragmentStateAdapter
import com.barryzeha.pomodoroapp.view.HistoryFragment
import com.barryzeha.pomodoroapp.view.MainFragment
import com.barryzeha.pomodoroapp.view.SettingsFragment

/****
 * Project PomodoroApp
 * Created by Barry Zea H. on 16/10/2022
 * Copyright (c)  All rights reserved.
 ***/
class ViewPagerAdapter(fragmentManager:FragmentManager, lifecycle: Lifecycle): FragmentStateAdapter(fragmentManager, lifecycle) {
    override fun getItemCount(): Int {
        return 3
    }

    override fun createFragment(position: Int): Fragment {
        return when (position){
            0 ->{MainFragment()}
            1 ->{HistoryFragment()}
            2 ->{SettingsFragment()}
            else ->{MainFragment()}
        }
    }
}